node/tests/swamp/swamp.go

```diff
@@ -0,0 +1,151 @@
+package swamp
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"net"
+	"testing"
+
+	"github.com/ipfs/go-log/v2"
+	"github.com/libp2p/go-libp2p-core/host"
+	mocknet "github.com/libp2p/go-libp2p/p2p/net/mock"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/stretchr/testify/require"
+
+	"github.com/celestiaorg/celestia-node/core"
+	"github.com/celestiaorg/celestia-node/libs/keystore"
+	"github.com/celestiaorg/celestia-node/node"
+	"github.com/celestiaorg/celestia-node/node/p2p"
+)
+
+var blackholeIP6 = net.ParseIP("100::")
+
+// TODO(@Bidon15):
+// in case ConnectPeers from Swamp.Network is utilized to connect nodes between each other
+// then p2p_exchange.go line 55 close() will panic due to being closed from dial_sync.go(go-libp2p-swarm lib)
+// in func DialLock, which is calling ad.decref() that closes the channel
+// _, err = sw.Network.ConnectPeers(bridge.Host.ID(), light.Host.ID())
+// require.NoError(t, err)
+type Swamp struct {
```

Comment:
Maybe stupid question but why is this actually called swamp? Is this common in testing? my understanding is that this swamp is a mix of integration tests that leani a bit towards e2e tests.

Reply:
`it's all ogre now`

Reply:
> it's all ogre now
:rofl:

Reply:
No it's not. However, it is common to have the naming to be bees related like `hive` or `x-swarm`, so I opted for some Shrek's swamp 😸